src/main/groovy/com/morpheusdata/hyperv/HyperVCloudProvider.groovy

```diff
@@ -645,6 +720,106 @@ class HyperVCloudProvider implements CloudProvider {
 		return 'Hyper-V'
 	}
 
+	private updateHypervisorStatus(server, status, powerState, msg) {
+		log.debug ("server: {}, status: {}, powerState: {}, msg: {}", server, status, powerState, msg)
+		if (server.status != status || server.powerState != powerState) {
+			server.status = status
+			server.powerState = powerState
+			server.statusDate = new Date()
+			server.statusMessage = msg
+			context.services.computeServer.save(server)
+		}
+	}
+
+	private Map refreshCache(opts, node) {
+		log.debug ("refreshCache: opts: ${opts}")
```

Comment:
This method is redundant. The results of `listVirualMachines()` in the parent method and the parent method already loads the same hypervOpts. Just move this up to the parent method and remove `refreshCache`

Reply:
review comment changes added as suggested